django/db/models/functions.py

```diff
@@ -182,13 +182,13 @@ def __init__(self, expression, pos, length=None, **extra):
             expressions.append(length)
         super(Substr, self).__init__(*expressions, **extra)
 
-    def as_sqlite(self, compiler, connection):
+    def as_sqlite(self, compiler, connection, **kwargs):
         self.function = 'SUBSTR'
-        return super(Substr, self).as_sql(compiler, connection)
+        return super(Substr, self).as_sql(compiler, connection, **kwargs)
 
-    def as_oracle(self, compiler, connection):
+    def as_oracle(self, compiler, connection, **kwargs):
         self.function = 'SUBSTR'
-        return super(Substr, self).as_sql(compiler, connection)
+        return super(Substr, self).as_sql(compiler, connection, **kwargs)
```

Comment:
This and one above, replace `self.function = ..` with `function=...` in as_sql method call.